addons/storyshots/src/test-bodies.js

```diff
@@ -1,12 +1,30 @@
+import path from 'path';
 import renderer from 'react-test-renderer';
 import shallow from 'react-test-renderer/shallow';
+import 'jest-specific-snapshot';
 
-export const snapshotWithOptions = options => ({ story, context }) => {
+function getRenderedTree(story, context, options) {
   const storyElement = story.render(context);
-  const tree = renderer.create(storyElement, options).toJSON();
+  return renderer.create(storyElement, options).toJSON();
+}
+
+function getSnapshotFileName(context) {
+  const fileName = context.storyFileName || __filename;
```

Comment:
Yeah this will just be the file `test-bodies.js` right? I'm not sure what a good default is but perhaps it should be figured out by the `jest` plugin?